Document the astro:beforeload event (#3837)
withastro/docs: src/content/docs/en/guides/view-transitions.mdx

  }, { once: true });
</script>
```

> 

Reviewer: Given that we now have `astro:load` *and* `astro:beforeload`, we probably do need a line or two about the difference, rather than just relying on people reading the description of each and working it out.

I’d suggest an `h3` before `astro:load`, e.g. `### Useful events` or something with one or two sentences describing the difference.

May also be good to update the first sentence under `astro:load`? That says “An event that fires on page navigation” but how is that different from the `beforeload` “after the new page replaces the old page”?

Author: Hm, is that common? It seems like it's just going to be a repeat of the sentences in each of the event descriptions. 

> May also be good to update the first sentence under astro:load?

yes, this one can be more specific.

Author: Updated the astro:load.

#### `astro:beforeload`

An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document`.

> This event is useful to restore any state on the DOM that needs to transfer over to the new page. If you are implementing dark mode support, this event can be used to restore state onto the new page like so:

Reviewer: Breaking up the generic “this is useful” and the specific example across paragraphs:

```suggestion
This event is useful to restore any state on the DOM that needs to transfer over to the new page.

For example, if you are implementing dark mode support, this event can be used to restore state across page loads:
```

```astro

> <script>
> 	const toggle = () => {
> 		if(localStorage.darkMode) {
> 			document.documentElement.dataset.dark = '';
> 		}
> 	};
> 
>   // MPA nav
> 	toggle();
>   // ViewTransitions nav
> 	document.addEventListener('astro:beforeload', toggle);
> </script>
> <style is:global>
> 	html[data-dark] body {
> 		background: black;
> 		color: white;
> 	}
> </style>

Reviewer: I think we can do some renaming here to clarify what’s happening, plus drop the `<style>` block which isn’t crucial to the example:

```suggestion
<script>
	const setDarkMode = () => {
		if (localStorage.darkMode) {
			document.documentElement.dataset.dark = '';
		}
	};

  // Runs on initial navigation
	setDarkMode();
  // Runs on view transitions navigation
	document.addEventListener('astro:beforeload', setDarkMode);
</script>
```
